miner/src/service_transaction_checker.rs

```diff
@@ -37,34 +37,34 @@ pub trait ContractCaller {
 /// Service transactions checker.
 #[derive(Default)]
 pub struct ServiceTransactionChecker {
-	contract: Mutex<Option<Contract>>,
+	contract: service_transaction::ServiceTransaction,
+	contract_address: Mutex<Option<Address>>,
 }
 
 impl ServiceTransactionChecker {
 	/// Try to create instance, reading contract address from given chain client.
 	pub fn update_from_chain_client(&self, client: &ContractCaller) {
-		let mut contract = self.contract.lock();
-		if contract.is_none() {
-			*contract = client.registry_address(SERVICE_TRANSACTION_CONTRACT_REGISTRY_NAME)
-				.and_then(|contract_addr| {
-					trace!(target: "txqueue", "Configuring for service transaction checker contract from {}", contract_addr);
-
-					Some(Contract::new(contract_addr))
-				})
+		let mut contract_address = self.contract_address.lock();
```

Comment:
The whole method is useless imho. just call registrar on `check` every time (since the address of the contract may change in the meantime) or call once at the start and memoize if we don't care that it might change.

With this approach we only support a case where the contract address is set to the registry, but we started earlier. IMHO it's a weird / too specific.